### ISSUE TITLE
GetxController onInit and onReady methods called twice

### ISSUE THREAD

**Issue by na2axl** [jonataslaw/getx#571]
**Describe the bug**
It seems that for a specific case, `onInit` and `onReady` callback are called twice, the first time when the controller is added using `Get.put` (which is normal), and the second time when the controller is used in a `GetX` widget (which I think the initialization must be avoided if already done). This lead for me to create animation controllers and run HTTP queries two times...

**To Reproduce**
1. Create a simple controller
2. Call Get.put() with this controller, assign a tag to it
3. Create a page using this controller, instantiate the controller using a custom tag and then use it in `GetX` widget call, as the `init` parameter
4. The controller is initialized the first time when calling `Get.put`, and the second time in the `initState` of the `GetX` widget

**Expected behavior**
get should know when it is necessary to initialize a controller.

**Flutter Version:**
1.20.0

**Getx Version:**
3.8.0

**Describe on which device you found the bug:**
Samsung SM A7000

**Minimal reproduce code**
```dart
class PageController extends GetxController {
  final RxString _text = "Example".obs;
  String get text => _text.value;
}

class PageBindings extends Bindings {
  final String tag;

  PageBindings({
    @required String tag,
  });

  @override
  void dependencies() => Get.lazyPut(() => PageController(), tag: this.tag);
}

class Page extends StatelessWidget {
  final PageController controller;

  Page({
    Key key,
    @required String tag,
  }) : controller = Get.find(tag: tag),
      super(key: key);

  @override
  Widget build() => Container(
    child: GetX(
      init: controller,
      global: false,
      builder: (_) => Text(controller.text),
    ),
  );
}
```

**Comment by jonataslaw** [comment]
Hi, initially I need to explain what you are doing:

```dart
  @override
  void dependencies() => Get.lazyPut(() => PageController(), tag: this.tag);
```

Here you have created a dependency of type PageController, with a tag.

```dart
  @required String tag,
  }) : controller = Get.find(tag: tag),
```
At that time the controller was created by calling Get.find() (Controller PageController, tag: "your tag" )

```dart
GetX(
      init: controller,
      global: false,
```
Here you used global = false.
This will literally create a new internal instance and pass it on to the children.
When creating an inner class, onInit is called.

```dart
class PageController extends GetxController {
  final RxString _text = "Example".obs;
  String get text => _text.value;
}

class PageBindings extends Bindings {
  final String tag;

  PageBindings({
    @required String tag,
  });

  @override
  void dependencies() => Get.lazyPut(() => PageController(), tag: this.tag);
}

class Page extends StatelessWidget {
  final PageController controller;

  Page({
    Key key,
    @required String tag,
  }) : controller = Get.find(tag: tag),
      super(key: key);

  @override
  Widget build() => Container(
    child: GetX(
      builder: (_) => Text(controller.text),
    ),
  );
}

```
This will cause it to be called only once.

This behavior is expected, since you can have an inner class in addition to the global class.
So, closing this.

**Comment by na2axl** [comment]
Hi @jonataslaw, thank you everything is clear now, I've used `init` and `global: false` because I fear that the `GetX` widget will be unable to find the correct controller because I'm using a `tag`.

Maybe a new `tag` field in the `GetX` widget will be useful on this case ? So we will not obliged to create an external controller directly in the `StatelessWidget` but we will directly pass the required `tag` to the `GetX` widget:

```dart
class Page extends StatelessWidget {
  final String tag;

  Page({
    Key key,
    @required this.tag,
  }) : super(key: key);

  @override
  Widget build() => Container(
    child: GetX(
      tag: tag,
      builder: (controller) => Text(controller.text),
    ),
  );
}
```

What do you think @jonataslaw ?

**Comment by jonataslaw** [comment]
Try
```dart
GetX<PageController>(
      tag: tag,
      builder: (controller) => Text(controller.text),
    ),
```
